parsl/executors/high_throughput/executor.py

```diff
@@ -573,19 +575,30 @@ def scale_out(self, blocks=1):
         Raises:
              NotImplementedError
         """
+        if not self.provider:
+            raise (ScalingFailed("No execution provider available"))
         r = []
         for i in range(blocks):
             external_block_id = str(len(self.blocks))
-            launch_cmd = self.launch_cmd.format(block_id=external_block_id)
-            internal_block = self.provider.submit(launch_cmd, 1)
-            logger.debug("Launched block {}->{}".format(external_block_id, internal_block))
-            if not internal_block:
-                raise(ScalingFailed(self.provider.label,
-                                    "Attempts to provision nodes via provider has failed"))
-            r.extend([external_block_id])
-            self.blocks[external_block_id] = internal_block
+            try:
+                self.blocks[external_block_id] = self._launch_block(external_block_id)
+                r.append(external_block_id)
+            except Exception as ex:
+                self._fail_job_async(external_block_id,
+                                     "Failed to start block {}: {}".format(external_block_id, ex))
         return r
 
+    def _launch_block(self, external_block_id: str) -> Any:
+        if self.launch_cmd is None:
+            raise ScalingFailed(self.provider.label, "No launch command")
+        launch_cmd = self.launch_cmd.format(block_id=external_block_id)
+        internal_block = self.provider.submit(launch_cmd, 1)
+        logger.debug("Launched block {}->{}".format(external_block_id, internal_block))
+        if not internal_block:
+            raise(ScalingFailed(self.provider.label,
```

Comment:
This PR has the infrastructure for error handling. It is only fully implemented at this moment in the local provider. Providers should, in their submit() method, report errors through exceptions rather than magical return values. However, updates to all providers to do so are not within the scope of this PR.